targets/TARGET_Analog_Devices/TARGET_ADUCM302X/TARGET_ADUCM3029/api/us_ticker.c

```diff
@@ -0,0 +1,347 @@
+    	tmrpend0 = NVIC_GetPendingIRQ(adi_tmr_interrupt[ADI_TMR_DEVICE_GP1]);
+  																    // Check if there is a pending interrupt for timer 1
+
+        __DMB();                                                    // memory barrier: read GP0 before GP1
+
+    	tmrcnt0 = adi_tmr_registers[ADI_TMR_DEVICE_GP0]->CURCNT;    // to minimize skew, read both timers manually
+
+    	__DMB();                                                    // memory barrier: read GP0 before GP1
+
+    	tmrcnt1 = adi_tmr_registers[ADI_TMR_DEVICE_GP1]->CURCNT;    // read both timers manually
+
+    	totaltmr0 = tmrcnt0;        								// expand to u32 bits
+    	totaltmr1 = tmrcnt1;        								// expand to u32 bits
+
+    	tmrcnt0 &= 0xff00u;
+    	tmrcnt1 <<= 8;
+
+        __DMB();
+
+    	uc1 = *ucptr;												// Read Upper_count
+
+    	tmrpend1 = NVIC_GetPendingIRQ(adi_tmr_interrupt[ADI_TMR_DEVICE_GP1]);
+    				// Check for a pending interrupt again.  Only leave loop if they match
+
+        NVIC_EnableIRQ(adi_tmr_interrupt[ADI_TMR_DEVICE_GP1]);		// enable interrupt on every loop to allow TMR1 interrupt to run
+    } while ((tmrcnt0 != tmrcnt1) || (tmrpend0 != tmrpend1));
+
+	totaltmr1 <<= 8;                 // Timer1 runs 256x slower
+	totaltmr1 += totaltmr0 & 0xffu;  // Use last 8 bits of Timer0 as it runs faster
+									 // totaltmr1 now contain 24 bits of significance
+
+	if (tmrpend0) {					 // If an interrupt is pending, then increment local copy of upper count
+		uc1++;
+	}
+
+	uint64_t Uc = totaltmr1;         // expand out to 64 bits unsigned
+	Uc += ((uint64_t) uc1) << 24;    // Add on the upper count to get the full precision count
+
+	 	 	 	 	 	 	 	 	 // Divide Uc by 26 (26MHz converted to 1MHz) todo scale for other clock freqs
+
+	Uc *= 1290555u;                  // Divide total(1/26) << 25
+	Uc >>= 25;						 // shift back.  Fixed point avoid use of floating point divide.
+									 // Compiler does this inline using shifts and adds.
+
+	return Uc;
+}
+
+
+static void calc_event_counts(uint32_t timestamp)
+{
+    uint32_t calc_time, blocks, offset;
+	uint64_t aa;
+
+    calc_time = get_current_time();
+    offset = timestamp - calc_time; // offset in useconds
+
+    if (offset > 0xf0000000u)       // if offset is a really big number, assume that timer has already expired (i.e. negative)
+    	offset = 0u;
+
+    if (offset > 10u) {             // it takes 10us to user timer routine after interrupt.  Offset timer to account for that.
+    	offset -= 10u;
+    } else
+    	offset = 0u;
+
+    aa = (uint64_t) offset;
+    aa *= 26u;                      // convert from 1MHz to 26MHz clock. todo scale for other clock freqs
+
+    blocks = aa >> 7;
+    blocks++;                       // round
+
+	largecnt = blocks>>1;           // communicate to event_timer() routine
+}
+
+static void event_timer()
+{
+    if (largecnt) {
+    	uint32_t cnt = largecnt;
+
+    	if (cnt > 65535u) {
+    		cnt = 0u;
```

Comment:
There are misaligned lines in this file, 201, 199, 190, 206, 225, 329, 152 please  check this file . Any tabs used or just wrong spaces?

line 201 should be (missing `{` `}` )
```
} else {
   //code
}
```

Reply:
By the globals, did you mean the variables stdio_uart_inited and stdio_uart? They are global because they are referenced from mbed_board.c.
In terms of the misaligned lines in us_ticker, they look OK in my editors. I have added the missing brackets.
The test results are attached here:
[ad3029-15112017.zip](https://github.com/ARMmbed/mbed-os/files/1474546/ad3029-15112017.zip)

Reply:
@0xc0170  Is anything else needed for your approval on this PR?  Just noticed that the requested test results were posted in this code comment.